x-pack/plugins/ai_infra/product_doc_base/server/services/package_installer/utils/zip_archive.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import yauzl from 'yauzl';
+
+export interface ZipArchive {
+  hasEntry(entryPath: string): boolean;
+  getEntryPaths(): string[];
+  getEntryContent(entryPath: string): Promise<Buffer>;
+  close(): void;
+}
+
+export const openZipArchive = async (archivePath: string): Promise<ZipArchive> => {
+  return new Promise<ZipArchive>((resolve, reject) => {
+    const entries: yauzl.Entry[] = [];
+    yauzl.open(archivePath, { lazyEntries: true, autoClose: false }, (err, zipFile) => {
+      if (err || !zipFile) {
+        return reject(err ?? 'No zip file');
```

Comment:
Some utility to open a zip archive in memory, but without loading all the uncompressed entry at once, but only per-entry and on demand.